### DISCUSSION TITLE
Token Id

### DISCUSSION THREAD

**Discussion by abraraltaf92** [GetStream/stream-chat-flutter#384] [Q&A] [answered]
How to generate a token ID in dart for production use?

**Comment by imtoori** [comment, marked as answer]
As you can read in the [website documentation](https://getstream.io/chat/docs/flutter-dart/tokens_and_authentication/?language=dart&q=token) you usually should send this token from your backend to the client-side when a user registers or logs in.

You can use a `TokenProvider` to let the StreamChatClient refresh the token automatically once it expires

For quick development, it is also possible to disable token authentication and use client-side generated tokens using `StreamChatClient.devToken(userId)`

**Reply by abraraltaf92** [reply]
I didn't get the dart code for token provider in the website documentation .

**Reply by itsarvinddev** [reply]
Hey @abraraltaf92 have you found any way to generate token id in Dart?
